apps/journeys-admin/src/components/TemplateLibrary/TemplateLibrary.tsx

```diff
@@ -1,35 +1,58 @@
 import { ReactElement } from 'react'
+import { useTranslation } from 'react-i18next'
 import Container from '@mui/material/Container'
-import { GetPublishedTemplates_journeys as Journey } from '../../../__generated__/GetPublishedTemplates'
+import { GetJourneys_journeys as Journey } from '../../../__generated__/GetJourneys'
+import { GetPublishedTemplates_journeys as Template } from '../../../__generated__/GetPublishedTemplates'
 import { TemplateCard } from '../TemplateCard'
+import { ContactSupport } from '../ContactSupport'
 
 interface TemplateLibraryProps {
+  isPublisher?: boolean
   journeys?: Journey[]
+  templates?: Template[]
 }
 
 export function TemplateLibrary({
-  journeys
+  isPublisher = false,
+  journeys,
+  templates
 }: TemplateLibraryProps): ReactElement {
+  const { t } = useTranslation('apps-journeys-admin')
+
+  // journey == null is journey loading
+  const showLibrary = journeys == null || journeys?.length > 0 || isPublisher
```

Comment:
This is the bit thats causing the flash. Can we move this check out into its own check? 

```suggestion
  const showLibrary = journeys == null || journeys?.length > 0
```

Reply:
oops this is not the problem